### ISSUE TITLE
Enum value Delivered not present in SDK?

### ISSUE THREAD

**Issue by brandonewhite** [lob/lob-java#324]
When calling a Check `get`, an execption is thrown:

`java.lang.IllegalArgumentException: Unexpected value 'Delivered'
	at com.lob.model.TrackingEventNormal$NameEnum.fromValue(TrackingEventNormal.java:141)
	at com.lob.model.TrackingEventNormal$NameEnum$Adapter.read(TrackingEventNormal.java:153)
	at com.lob.model.TrackingEventNormal$NameEnum$Adapter.read(TrackingEventNormal.java:144)
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:130)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:221)
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:130)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:221)
	at com.google.gson.Gson.fromJson(Gson.java:991)
	at com.google.gson.Gson.fromJson(Gson.java:956)
	at com.google.gson.Gson.fromJson(Gson.java:905)
	at com.lob.api.JSON.deserialize(JSON.java:146)
	at com.lob.api.ApiClient.deserialize(ApiClient.java:821)
	at com.lob.api.ApiClient.handleResponse(ApiClient.java:1029)
	at com.lob.api.ApiClient.execute(ApiClient.java:951)
	at com.lob.api.client.ChecksApi.getWithHttpInfo(ChecksApi.java:416)
	at com.lob.api.client.ChecksApi.get(ChecksApi.java:392)`

This is present in versions 13.1.0 and 13.4.0

**Comment by BennyKitchell** [comment]
Hey @brandonewhite  

Thanks for bringing this to our attention I will investigate this tomorrow. In the meantime could you give me any more details that might help me debug this? Is it simply you were using `13.1.0` and using this enum, now it is not available in `13.4.0`?

**Comment by brandonewhite** [comment]
We recently updated to 13.1.0 and discovered the error when retrieving a check with a delivered status in the tracking events.  I'm guessing the event or status named "Delivered" is no longer present?  But it's the SDK that's throwing the error, we aren't specifically looking for values (we just output the events).  I tried the latest version 13.4.0 hoping it was fixed and saw the error is still thrown.  The error is not thrown when the check has not yet been delivered.

**Comment by BennyKitchell** [comment]
@brandonewhite  Thanks, I will look into this and get a fix out asap. Apologies for this. I expect to have a fix for by the end of next week, but that's not set in stone. In the meantime, could you use the TrackinEventDetails enum just to curcumvent the error until I can get a solid fix. I want to make sure to unblock you asap.

**Comment by brandonewhite** [comment]
Thanks @BennyKitchell - I don't think we can get around the exception at the moment.  We are only calling the "get" method on ChecksApi and the error is thrown.


From your SDK:
```
public Check get(String chkId) throws ApiException {
        try {
            ApiResponse<Check> localVarResp = getWithHttpInfo(chkId);
            return localVarResp.getData();
        } catch (ApiException e) {
            throw e;
        }
    }
```

**Comment by BennyKitchell** [comment]
@brandonewhite  I think the issue might be that the deliverable string is capitalized, could you try changing `Delivered` ->  `delivered` and see if that fixes the issue? If that's the case, I can try and implement a fix on my side that makes that change before comparison.

**Comment by brandonewhite** [comment]
@BennyKitchell The exception happens without referencing any values/variables/enums.  Just calling the "get" Check throws the exception.  I can try to roll back to a previous version to try and see when the issue was introduced.

**Comment by BennyKitchell** [comment]
@brandonewhite  I am unable to replicate this locally, is there anyway you can send me the raw object (if you could fetch the check using our api via postman or curl). Alternatively, if you could send me the check id, I can find it and see if I can find the issue.

**Comment by brandonewhite** [comment]
@BennyKitchell I've tracked this down a little more.  The generated Java class (I believe these Java objects are auto-generated using Open API tools) `com.lob.model.TrackingEventNormal` has a `public enum NameEnum` declaration that is missing the value of `DELIVERED("Delivered")` (JSON payload of tracking events from a raw request is below the enum code block).

`public enum NameEnum {
    IN_TRANSIT("In Transit"),
    
    IN_LOCAL_AREA("In Local Area"),
    
    PROCESSED_FOR_DELIVERY("Processed for Delivery"),
    
    RE_ROUTED("Re-Routed"),
    
    RETURNED_TO_SENDER("Returned to Sender"),
    
    MAILED("Mailed");

    private String value;

    NameEnum(String value) {
      this.value = value;
    }

    public String getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    public static NameEnum fromValue(String value) {
      for (NameEnum b : NameEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    public static class Adapter extends TypeAdapter<NameEnum> {
      @Override
      public void write(final JsonWriter jsonWriter, final NameEnum enumeration) throws IOException {
        jsonWriter.value(enumeration.getValue());
      }

      @Override
      public NameEnum read(final JsonReader jsonReader) throws IOException {
        String value =  jsonReader.nextString();
        return NameEnum.fromValue(value);
      }
    }
  }`

`"tracking_events": [
      {
        "id": "evnt_39b5463882bf73076087",
        "type": "normal",
        "name": "In Transit",
        "details": null,
        "location": "60499",
        "time": "2023-02-09T00:26:43.000Z",
        "date_created": "2023-02-09T04:37:30.406Z",
        "date_modified": "2023-02-09T04:37:30.406Z",
        "object": "tracking_event"
      },
      {
        "id": "evnt_26a9a22e0e00273ecae5",
        "type": "normal",
        "name": "In Transit",
        "details": null,
        "location": "60499",
        "time": "2023-02-09T04:21:02.000Z",
        "date_created": "2023-02-09T05:39:07.181Z",
        "date_modified": "2023-02-09T05:39:07.181Z",
        "object": "tracking_event"
      },
      {
        "id": "evnt_5c20cbea5d3edf50eca7",
        "type": "normal",
        "name": "In Local Area",
        "details": null,
        "location": "72231",
        "time": "2023-02-11T13:11:09.000Z",
        "date_created": "2023-02-11T14:39:39.334Z",
        "date_modified": "2023-02-11T14:39:39.334Z",
        "object": "tracking_event"
      },
      {
        "id": "evnt_753aece0cb928856362b",
        "type": "normal",
        "name": "Processed for Delivery",
        "details": null,
        "location": "72231",
        "time": "2023-02-12T10:41:55.000Z",
        "date_created": "2023-02-12T11:40:59.612Z",
        "date_modified": "2023-02-12T11:40:59.612Z",
        "object": "tracking_event"
      },
      {
        "id": "evnt_2f36427c23074d1766d4",
        "type": "normal",
        "name": "Delivered",
        "details": null,
        "location": "72204",
        "time": "2023-02-13T16:00:50.000Z",
        "date_created": "2023-02-13T19:35:27.262Z",
        "date_modified": "2023-02-13T19:35:27.262Z",
        "object": "tracking_event"
      }
    ]`

**Comment by BennyKitchell** [comment]
@brandonewhite AH man I see the issue, that was a problem with our  spec that was unnoticed. Thanks so much for calling this out. I will have a new version out hopefully by EoD. At the latest you will have it monday morning. Thanks for sticking with me through this.

**Comment by brandonewhite** [comment]
All good - glad you could track it down.  Will the update be available on 13.1.0 or only > 13.4.0?

**Comment by BennyKitchell** [comment]
I released it in a new version as [13.4.2](https://central.sonatype.com/artifact/com.lob/lob-java/13.4.2)
